### PR TITLE
Changed AppDetailsPage IsXXX to a collection of readonly checkboxes

### PR DESCRIPTION
**Pull request by andreww-msft**
As part of this, created a new FrameworkTypes model, and AppRuntimeInfo now has an ObservableCollection of these. This should make it easier to add new framework types later (I added Python). In the UI, each item is represented by a readonly CheckBox in a WrapPanel, so that they flow when the window is sized. Standard CheckBox doesn't support readonly, so I added a custom ReadOnlyCheckBox control.

Also simplified the visibility of the problem fields in the AppDetailsPage - these are all now within a grid, so we only have to set visibility on the grid instead of on each field (should also improve perf very slightly).

Sorted the module list by name. Added a GridSplitter to the Modules page, so that the user can size the width of the list.

Also, when the user closes the Bar window, we save the expanded size to user settings (if not maximized).

Deleted some redundant fields and strings.

### CODE REVIEW

File: tools/PI/DevHome.PI/Models/FrameworkType.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        get; set;
+    }
+
+    public string Name
+    {
+        get; set;
+    }
+
+    [ObservableProperty]
+    private bool status;
```
Comment (dhoehna) [inline tools/PI/DevHome.PI/Models/FrameworkType.cs:23]: What does status represent?

File: tools/PI/DevHome.PI/PIApp.xaml [hunk 2]
```diff
@@ -27,6 +27,7 @@
             </Style>
             <StackLayout x:Name="ExternalToolsHorizontalLayout" Orientation="Horizontal" />
             <StackLayout x:Name="ExternalToolsVerticalLayout" Orientation="Vertical" />
+
```
Comment (dhoehna) [inline tools/PI/DevHome.PI/PIApp.xaml:30]: NIT: extra line.

File: tools/PI/DevHome.PI/ViewModels/AppDetailsPageViewModel.cs [hunk 3]
```diff
@@ -63,51 +62,15 @@ public void UpdateTargetProcess(Process process)
                     {
                         AppInfo.MainModuleFileName = targetProcess.MainModule.FileName;
                         uint binaryTypeValue;
+
+                        // TODO GetBinaryType only distinguishes x86 from x64. It doesn't allow for ARM or ARM64.
```
Comment (dhoehna) [inline tools/PI/DevHome.PI/ViewModels/AppDetailsPageViewModel.cs:66]: Does DevHome have an issue tracking this potential work?

**Comment by dhoehna** [comment]
> support readonly
Checkboxes have `IsEnabled`.    That does make the control read-only.  Did you need other behavior besides this?

Can you please put a video of the changes in the PR?

File: tools/PI/DevHome.PI/Models/AppRuntimeInfo.cs [hunk 4]
```diff
@@ -62,4 +40,47 @@ public partial class AppRuntimeInfo : ObservableObject
 
     [ObservableProperty]
     private Visibility visibility = Visibility.Visible;
+
+    public ObservableCollection<FrameworkType> FrameworkTypes { get; } = [];
+
+    public AppRuntimeInfo()
+    {
+        // Note these are in alphabetical order of Name.
+        FrameworkTypes.Add(new FrameworkType("Avalonia.Base.dll", "Avalonia"));
```
Comment (jaholme) [inline tools/PI/DevHome.PI/Models/AppRuntimeInfo.cs:49]: Are these names localizable?

File: tools/PI/DevHome.PI/Models/AppRuntimeInfo.cs [hunk 5]
```diff
@@ -62,4 +40,47 @@ public partial class AppRuntimeInfo : ObservableObject
...
+        FrameworkTypes.Add(new FrameworkType("Microsoft.WinUI.dll", "WinUI"));
+        FrameworkTypes.Add(new FrameworkType("PresentationFramework.dll", "WPF"));
+    }
+
+    public void CheckFrameworkTypes(string moduleName)
+    {
+        foreach (var item in FrameworkTypes)
+        {
+            // Skip if already matched.
+            if (item.Status == true)
```
Comment (zadesai) [inline tools/PI/DevHome.PI/Models/AppRuntimeInfo.cs:65]: With a new targetprocess, we always create a new AppRuntimeInfo object. In what scenario will we have already matched the frameworktype?

File: tools/PI/DevHome.PI/Models/FrameworkType.cs (hunk 1 above)
Comment (andreww-msft) [reply tools/PI/DevHome.PI/Models/FrameworkType.cs:23]: FrameworkType is used to report which framework types the app uses, so status==true means it does support this type, and status==false means it does not support this type.

File: tools/PI/DevHome.PI/PIApp.xaml (hunk 2 above)
Comment (andreww-msft) [reply tools/PI/DevHome.PI/PIApp.xaml:30]: Yes, leaving blank lines between different functional definitions seems reasonable, no?

File: tools/PI/DevHome.PI/ViewModels/AppDetailsPageViewModel.cs (hunk 3 above)
Comment (andreww-msft) [reply tools/PI/DevHome.PI/ViewModels/AppDetailsPageViewModel.cs:66]: Yes.

File: tools/PI/DevHome.PI/Models/AppRuntimeInfo.cs (hunk 4 above)
Comment (andreww-msft) [reply tools/PI/DevHome.PI/Models/AppRuntimeInfo.cs:49]: No.

File: tools/PI/DevHome.PI/Models/AppRuntimeInfo.cs (hunk 5 above)
Comment (andreww-msft) [reply tools/PI/DevHome.PI/Models/AppRuntimeInfo.cs:65]: We're doing the check for every module. For cases where the match pattern is a partial match, we might already have matched say mfc123.dll and now we're seeing say mfc456.dll.

**Review by jaholme — APPROVED** [review]

**Comment by andreww-msft** [comment]
AppDetails-checkboxes

**Review by zadesai — APPROVED** [review]

File: tools/PI/DevHome.PI/Models/FrameworkType.cs (hunk 1 above)
Comment (dhoehna) [reply tools/PI/DevHome.PI/Models/FrameworkType.cs:23]: Can Status be renamed to IsTypeSupported to make it more descriptive?

Comment (andreww-msft) [reply tools/PI/DevHome.PI/Models/FrameworkType.cs]: Good idea. Fixed in new change.

File: tools/PI/DevHome.PI/PIApp.xaml (hunk 2 above)
Comment (andreww-msft) [reply tools/PI/DevHome.PI/PIApp.xaml:31]: Sorry, I meant "functional" not "function" - different definitions in the XAML that serve different functional purposes. Anyway, I've added another blank after the </Style> closing tag, as you suggested.

**Pull request merged**